src/llm_vm/completion/optimize.py

```diff
@@ -216,14 +215,29 @@ def complete_delay_train(self, stable_context, dynamic_prompt, c_id = None, **kw
             def promiseCompletion():
                 best_completion = self.call_big(prompt, **kwargs)
                 def actual_train(use_completion = None):
+                    def data_synthesis(prompt, response, **kwargs):
+                        final_prompt = '{"prompt": "' +prompt+'"  , "response": "' +response+'" }'+'\nGenerate 100 more JSONS each with a prompt and response field like the given one. The content of the prompt and response fields must be similar to the given JSON. Seperate each JSON with a ,.'
+                        data=self.call_big(final_prompt , kwargs)
+                        datapoints = []
+                        data = '{ "datapoints": ['+data+']}'
+                        print(data)
+                        data = json.loads(data)
+                        for i in data["datapoints"]:
+                            datapoints.append((i["prompt"],i["response"]))
+                        return datapoints
+
                     train_completion = best_completion if use_completion is None else use_completion
-                    new_datapoint = (dynamic_prompt.strip(), train_completion)
+                    new_datapoint = (prompt.strip(), train_completion)
                     self.storage.add_example(c_id, new_datapoint)
+                    for j in data_synthesis(prompt,best_completion):
+                        self.storage.add_example(c_id, j)
```

Comment:
@abhigya-sodani Won't this lead to polynomial growth of the data in storage synthesis?

Reply:
I do think it's important to assign a variable to generated dataset, have some basic checks before adding it to storage?

Reply:
> @abhigya-sodani Won't this lead to polynomial growth of the data in storage synthesis?

I am just adding all generated examples for finetuning, do we want to clear out the storage after finetuning is done to prevent the polynomial growth?